test/evp_test.c

```diff
@@ -2329,17 +2329,25 @@ static int check_test_error(EVP_TEST *t)
  */
 static int run_test(EVP_TEST *t)
 {
+    int r;
+
     if (t->meth == NULL)
         return 1;
     t->s.numtests++;
     if (t->skip) {
         t->s.numskip++;
     } else {
         /* run the test */
-        if (t->err == NULL && t->meth->run_test(t) != 1) {
-            TEST_info("%s:%d %s error",
-                      t->s.test_file, t->s.start, t->meth->name);
-            return 0;
+        if (t->err == NULL) {
+            if (t->expected_err != NULL)
+                test_suppress_error_diagnostics(1);
+            r = t->meth->run_test(t);
+            test_suppress_error_diagnostics(0);
```

Comment:
Some of the run_test functions do return 0 on some errors...  Should their diagnostics be supressed?  I think not!

I'm not sure why some of them return 1 on error, but perhaps it has to do with the class of error (i.e. if it's considered fatal or not)...

Reply:
This PR is aiming to quiet the negative tests in the output.  I.e. I'm assuming that all of the tests pass (both negative and positive), how can we reduce the noise?

I'm not sure how to best resolve this.  Returning zero means an error occurred but it is too late to not suppress any output.  We could, perhaps, cache output in a memory BIO and clear it on success.  That seems overly complicated and error prone to me.  I might, instead, add an extra keyword in the EVP input files to indicate that they should suppress output in expectation of a failure.  However, there is such an option already (the expected error) which is what I'm conditioning on currently.  If an unexpected error occurs when an error is expected, the `evp_test` wrapper will explain what it was in general terms but will omit any specific details.  If no errors occurs, when one is expected, there wasn't any useful output to be had so it won't be missed.

Better might be a command line option to `evp_test` which disables the suppression of output so one can reproduce the failure without censorship.  Even better would be a global command line (or environmental) option to the test framework to stop the suppression.  I did have a quick look into adding a command line option to enforce all output, it isn't trivial in the current test framework.  I'd be willing to give it a try, although after 1.1.1 would be prudent.  I've always planned on having global command line options but the current framework isn't quite ready for them just yet.